dot/core/service.go

```diff
@@ -346,6 +348,9 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 		return nil
 	}
 
+	s.storageState.Lock()
```

Comment:
For the storageState locks in this file, I had observed that with the locks the errors I was getting around the timestamp issue would go away. However, it doesn't permanently fix it and since @kishansagathiya has been investigating this issue maybe i remove these locks (as long as it still works ofc)?